Spigot-Server-Patches/0131-Optimize-getType.patch

```diff
@@ -0,0 +1,186 @@
+From db132f197f8d483beb0faaf21739a319643c60f1 Mon Sep 17 00:00:00 2001
+From: Techcable <Techcable@techcable.net>
+Date: Wed, 6 Apr 2016 09:19:00 -0700
+Subject: [PATCH] Optimize getType
+
+Removes mojang's new 'DataBits', which are stupidly inneficent
+
+Hopefully improve inlining
+
+diff --git a/src/main/java/net/minecraft/server/DataPaletteBlock.java b/src/main/java/net/minecraft/server/DataPaletteBlock.java
+index e29298e..c6e8718 100644
+--- a/src/main/java/net/minecraft/server/DataPaletteBlock.java
++++ b/src/main/java/net/minecraft/server/DataPaletteBlock.java
+@@ -2,11 +2,13 @@ package net.minecraft.server;
+ 
+ public class DataPaletteBlock implements DataPaletteExpandable {
+ 
++
+     private static final DataPalette d = new DataPaletteGlobal();
+     protected static final IBlockData a = Blocks.AIR.getBlockData();
+-    protected DataBits b;
++    // protected DataBits b; // Paper - nope
+     protected DataPalette c;
+     private int e = 0;
++    private final short[] data = new short[4096]; // Paper
```

Comment:
Though looking deeper, i see most the uses are new code. Id say it would be better (as in standard coding practices) to reference data.length instead of hard coding an expected length in.

Then the only reference will be the initializer.